torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -179,6 +179,12 @@ void ProcessGroupAgent::join() {
       SendWork(workerIds_[dst], Message({}, {}, MessageType::SHUTDOWN)));
   threadPool_.waitWorkComplete();
   listenerThread_.join();
+  // explicitly clean up python objects in PythonRpcHandler before destructing
+  // ProcessGroupAgent. Otherwise, in Python 3.5, when python program exits,
+  // the python objects in PythonRpcHandler could be cleaned up before
+  // destructing ProcessGroupAgent, then ProcessGroupAgent destructor will
+  // dec_ref deallocated python objects and crash.
```

Comment:
Question: do we know why `inc_ref` was not called properly in the first place? If `ProcessGroupAgent` is holding a reference as well, should the reference counting be consistent with that?